ddtrace/tracer/spancontext.go

```diff
@@ -290,10 +290,11 @@ func (t *trace) samplingPriority() (p int, ok bool) {
 	return t.samplingPriorityLocked()
 }
 
-func (t *trace) setSamplingPriority(p int, sampler samplernames.SamplerName) {
+// setSamplingPriority sets the sampling priority and returns true if it was modified.
+func (t *trace) setSamplingPriority(p int, sampler samplernames.SamplerName) bool {
 	t.mu.Lock()
 	defer t.mu.Unlock()
```

Comment:
Reviewing this it seems to me that the original issue comes from checking `c.trace.priority != nil && *c.trace.priority != float64(p)` outside of the highlighted mutex.

In general, the change seems safe.

Reply:
That is correct: this fix ensures we only read `t.priority` while holding the lock, which required returning the boolean from this function.